aten/src/ATen/native/mobile/cpu/Engine.h

```diff
@@ -0,0 +1,137 @@
+#pragma once
+
+#include <ATen/ATen.h>
+
+// Prefer not to use these functions directly.  Instead prefer to use the
+// corresponding mobile operators that this module registers with, and exposes
+// through, c10 whenever possible in order to decouple op creation from op
+// execution.  The underlying computation in many NN operators is performed in
+// phases, all of which do not need to run at the same frequency.  Decoupling op
+// creation from execution allows for one-time computations to be cached and
+// factored out resulting in time savings per calls to forward() - something
+// this API cannot do.  Furthermore, this API does not allow for fusion of
+// non-linear operators that, again, is something that the exposed c10 mobile
+// operators can handle.
+
+namespace at {
+namespace native {
+namespace mobile {
+namespace cpu {
+
+bool available();
+bool initialize();
+bool deinitialize();
+
+//
+// Add
+//
+
+bool use_add(
+    const Tensor& input1,
+    const Tensor& input2);
+
+Tensor& add(
```

Comment:
is there a reason you didn't follow the (native_functions) C++ standard naming of "add_out" here?  (And for the cases below).

Reply:
Right, I will modify all functions returning a Tensor reference to have _out in their name.

Reply:
Done.